src/popup/popup.js

```diff
@@ -1 +1,60 @@
-/* popup.js */
+import {
+  getAllExtensions,
+  areExtsBeingMonitored,
+  initMonitorAll,
+  stopMonitorAll,
+  viewActivityLogs,
+} from '../lib/ext-listen.js';
+
+async function init() {
+  const extensions = await getAllExtensions();
+  const startMonitorAllBtn = document.getElementById('startMonitorBtn');
+  const stopMonitorAllBtn = document.getElementById('stopMonitorBtn');
+  const viewActivityLogBtn = document.getElementById('actLogPage');
+  const monitorStatusText = document.getElementById('monitorStatus');
+
+  startMonitorAllBtn.addEventListener('click', async () => {
+    const monitorMsg = await initMonitorAll(extensions);
+    if (monitorMsg === 'ext-monitor-started') {
+      window.close();
+    } else {
+      alert("Extension monitoring couldn't be started");
+    }
+  });
+
+  stopMonitorAllBtn.addEventListener('click', async () => {
+    const monitorMsg = await stopMonitorAll();
+    if (monitorMsg === 'ext-monitor-stopped') {
+      window.close();
+    } else {
+      alert("Extension monitoring couldn't be stopped");
```

Comment:
Personally, I would prefer to avoid alert as the strategy used to show errors to the user.
Also, the popup is being just closed when the handler is successful, which I'm not sure is enough as a feedback to the user that the operation has been successfull (e.g. I would imagine that the user may just open the browserAction popup again to double-check that).

Instead, could we:
- when the handler has successfully started/stopped the monitor extensions: update the popup page (e.g. updating the UI as the user would expect when the button clicked handled the request successfully) and don't autoclose the popup
- when the handler did fail, update the popup page to show an error message to the user and don't autoclose the popup

Besides that, what if initMonitorAll or stopMonitorAll reject?

Reply:
Thanks for explaining in detail. I have added a throw statement [here](https://github.com/mozilla/extension-activity-monitor/pull/3/commits/eb10df7504be3bcf734b29a57b6da9df60658294#diff-aa663d2fedefb755dd0cf06fef1f44e2R46) and [here](https://github.com/mozilla/extension-activity-monitor/blob/eb10df7504be3bcf734b29a57b6da9df60658294/src/lib/ext-monitor.js#L95) if initMonitorAll fails.
I have added a throw statement [here](https://github.com/mozilla/extension-activity-monitor/blob/eb10df7504be3bcf734b29a57b6da9df60658294/src/lib/ext-monitor.js#L55) if stopMonitorAll fails. 

Since we are using `try {} catch {}` now. The rejects will be caught and the reject message will be displayed in the popup.